confidant_client/__init__.py

```diff
@@ -110,7 +120,7 @@ def __init__(
             'backoff': backoff,
             'assume_role': assume_role
         }
-        for key, val in args_config.iteritems():
+        for key, val in six.iteritems(args_config):
```

Comment:
just use `.items()`, the "performance benefits" of `six.iter*` are not worth the code smell:

http://lists.openstack.org/pipermail/openstack-dev/2015-June/066391.html

Reply:
All of these nits are on the output of `modernize` or existing code. I can fix some of them for now, but I'm not sure the effort is really worth it

Reply:
I was hoping you would take this advice in the rest of the `six.iter*` calls :)